src/dmd/traits.d

```diff
@@ -1417,19 +1417,28 @@ Expression semanticTraits(TraitsExp e, Scope* sc)
         auto o = (*e.args)[0];
         auto o1 = (*e.args)[1];
 
-        FuncDeclaration fd;
-        TypeFunction tf = toTypeFunction(o, fd);
-
         ParameterList fparams;
-        if (tf)
-            fparams = tf.parameterList;
-        else if (fd)
-            fparams = fd.getParameterList();
+
+        CallExp ce;
+        if (auto exp = isExpression(o))
+            ce = exp.isCallExp();
+
+        if(ce)
+            fparams = ce.f.getParameterList();
```

Comment:
Lines 1422-1427 can be rewritten to:

```d
if (auto exp = isExpression(o));
{
     if (auto ce = exp.isCallExo())
         fparams = ce.f.getParameterList();
}

Reply:
Done

Reply:
Oof, am I wrong? That change doesn't work, but should, right?

```d
if (auto exp = isExpression(o))
{
    if(auto ce = exp.isCallExp())
        fparams = ce.f.getParameterList();
}
```

```d
CallExp ce;
if (auto exp = isExpression(o))
    ce = exp.isCallExp();

if (ce)
    fparams = ce.f.getParameterList();
```

Reply:
Oh, of course, this can't be done because there's an else. I probably added whitespace by mistake and it confused me and you.